RecoTracker/LSTCore/src/alpaka/Triplet.h

```diff
@@ -427,7 +428,7 @@ namespace lst {
     rzChiSquared = 12 * (residual * residual) / (error * error * projection_missing2);
 
     //helix calculation returns NaN, use linear approximation
-    if (alpaka::math::isnan(acc, rzChiSquared) || circleRadius < 0) {
+    if (edm::isNotFinite(rzChiSquared) || circleRadius < 0) {
       float slope = (z_other - z1) / (r_other - r1);
 
       residual = (layer3 <= 6) ? ((z_target - z1) - slope * (r_target - r1)) : ((r_target - r1) - (z_target - z1) / slope);
```

Comment:
do I understand correctly that changes above (`z3 -> z_target`) mean that similar updates are needed here to keep in sync with the meaning of `z_target` and `z_other` (same for `r`) since some values are reshuffled depending on the layer combinations around L300 ?

Reply:
Thanks for pointing that out. I've updated the code so that we only use MD3 as the target, since it doesn't really make a difference here. There is no change in performance

Reply:
wasn't the `dz==0` case significant in the B456 T3s if MD3 was the target?

Reply:
For B456 (region 23), we use the original linear cut (L237-260), so this wouldn't be an issue